### ISSUE TITLE
CMake configuration issue on MacOS 11 / Arm64: ffs::ffs not found

### ISSUE THREAD

**Issue by franzpoeschel** [ornladios/ADIOS2#3655]
**Describe the bug**
On https://github.com/openPMD/openPMD-api/pull/1387, I am currently trying to update our Python Wheel to ADIOS2 v2.9.0. On MacOS 11 with Arm64, this results in a CMake configuration error:

```
2023-06-06T14:52:05.8244180Z CMake Error at CMakeLists.txt:9 (find_package):
2023-06-06T14:52:05.8244510Z   Found package configuration file:
2023-06-06T14:52:05.8244660Z 
2023-06-06T14:52:05.8244940Z     /usr/local/lib/cmake/adios2/adios2-config.cmake
2023-06-06T14:52:05.8245130Z 
2023-06-06T14:52:05.8245290Z   but it set adios2_FOUND to FALSE so package "adios2" is considered to be
2023-06-06T14:52:05.8245580Z   NOT FOUND.  Reason given by package:
2023-06-06T14:52:05.8245730Z 
2023-06-06T14:52:05.8245910Z   The following imported targets are referenced, but are missing: ffs::ffs
```

The full build log is: 
[build_log.txt](https://github.com/ornladios/ADIOS2/files/11674793/build_log.txt)

**To Reproduce**
The build script is found [here](https://github.com/openPMD/openPMD-api/blob/143e6aa440fde5465bf8a5ea5d86ea086ec7dbf5/library_builders.sh).

**Expected behavior**
Successful configuration

**Desktop (please complete the following information):**
The failing configuration is this part of our [`build.yml`](https://github.com/openPMD/openPMD-api/blob/143e6aa440fde5465bf8a5ea5d86ea086ec7dbf5/.github/workflows/build.yml):
```yml
          - os: macos-11
            arch: "arm64"
            env:
              CMAKE_OSX_ARCHITECTURES: "arm64"
              MACOSX_DEPLOYMENT_TARGET: 11.0
```

**Additional context**
I tried working around the issue by using external installs of dill, atl and ffs:

```bash
git clone https://github.com/ornladios/ADIOS2 
cd ADIOS2/thirdparty/atl/atl
mkdir build
cd build
cmake .. 
make install

# same for dill, ffs

cd ../../../..
mkdir build
cd build
cmake .. -DADIOS2_USE_EXTERNAL_FFS=ON -DADIOS2_USE_EXTERNAL_DILL=ON -DADIOS2_USE_EXTERNAL_ATL=ON -DADIOS2_USE_Fortran=OFF
```
But in that case, I get a similar error as above on all systems:
```
-- Configuring done
CMake Error at source/adios2/CMakeLists.txt:6 (add_library):
  Target "adios2_core" links to target "ffs::ffs" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or 
  an ALIAS target is missing?


-- Generating done 
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

[configure_log.txt](https://github.com/ornladios/ADIOS2/files/11674927/configure_log.txt)

**Following up**
Was the issue fixed? Please report back.

**Comment by chapouvalpin** [comment]
I have the exact same issue on Arch Linux.
[configure_and_build.log](https://github.com/ornladios/ADIOS2/files/11679933/configure_and_build.log)

**Comment by eisenhauer** [comment]
This seems quite similar to issue  #3309, which was supposedly fixed.  Can everyone report which version of CMake they are using?  @vicentebolea , tagging as you handled this earlier.

**Comment by vicentebolea** [comment]
@chapouvalpin you can disable the adios2-config script generation (which you do not need unless you want to link your app with ADIOS2 without CMake). 

Find the workaround in this command: https://github.com/ornladios/ADIOS2/issues/3348#issuecomment-1517084004

**Comment by franzpoeschel** [comment]
The workaround described at #3348 helps installing ADIOS2 successfully, but the same error occurs again later when trying to do `find_package(ADIOS2)` in our own CMake script.

**Comment by franzpoeschel** [comment]
We don't specify either of these options, neither as a CMake option or

**Comment by vicentebolea** [comment]
It might be worth to try to set either of those env var.

**Comment by vicentebolea** [comment]
This might have been resolved in #3760

**Comment by franzpoeschel** [comment]
Good to know, thank you!
@ax3l has in the meantime found a solution for https://github.com/openPMD/openPMD-api/pull/1387, though I'm not exactly sure what it was
I will close this for the moment and reopen in case the issue reappears
